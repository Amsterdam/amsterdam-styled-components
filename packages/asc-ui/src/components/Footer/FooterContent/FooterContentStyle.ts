import styled from '@datapunt/asc-core'
import { ShowHideTypes, showHide } from '../../../utils'

export interface Props extends ShowHideTypes {
  className?: string
  indent?: boolean
}

export default styled.div<Props>`
  display: flex;
  flex-direction: column;
  margin-left: ${({ indent }) => (indent ? '28px' : '0')};
  ${showHide}
`
