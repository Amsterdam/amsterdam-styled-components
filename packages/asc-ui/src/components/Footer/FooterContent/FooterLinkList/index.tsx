import styled from '@datapunt/asc-core'
import { color, svgFill } from '../../../../utils'
import LinkList, { LinkListItem } from '../../../LinkList'
import TypographyStyle from '../../../Typography/TypographyStyle'
import IconStyle from '../../../Icon/IconStyle'
import LinkStyle from '../../../Link/LinkStyle'

export const InverterdLinkList = styled(LinkList)`
  background-color: ${color('tint', 'level5')};
`

export const InverterdLinkListItem = styled(LinkListItem)`
  ${TypographyStyle} {
    color: ${color('tint', 'level1')};
  }

  ${IconStyle} {
    & > svg {
      ${svgFill('tint', 'level1')};
    }
  }

  ${LinkStyle} {
    color: ${color('tint', 'level1')};

    &:hover {
      color: ${color('tint', 'level1')};
      ${IconStyle} > svg {
        ${svgFill('tint', 'level1')};
      }
    }
  }
`
