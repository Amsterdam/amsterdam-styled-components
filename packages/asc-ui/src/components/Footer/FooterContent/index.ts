export { default } from './FooterContent'
export { default as FooterContentStyle } from './FooterContentStyle'
