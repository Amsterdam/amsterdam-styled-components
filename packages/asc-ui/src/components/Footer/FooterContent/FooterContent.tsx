import React from 'react'
import FooterContentStyle, {
  Props as FooterContentStyleProps,
} from './FooterContentStyle'

export interface Props extends FooterContentStyleProps {}

const FooterContent: React.FC<Props> = ({
  children,
  className,
  ...otherProps
}) => (
  <FooterContentStyle className={className} {...otherProps}>
    {children}
  </FooterContentStyle>
)

export default FooterContent
