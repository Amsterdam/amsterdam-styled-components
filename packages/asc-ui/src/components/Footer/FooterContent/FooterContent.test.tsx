import React from 'react'
import { render, cleanup } from '@testing-library/react'
import FooterContent from './FooterContent'

describe('FooterContent', () => {
  afterEach(cleanup)

  it('should render with and without the indent', () => {
    const { queryByTestId, rerender } = render(
      <FooterContent data-testid="test-id">Footer content</FooterContent>,
    )

    expect(queryByTestId('test-id')).toHaveStyleRule('display', 'flex')
    expect(queryByTestId('test-id')).toHaveStyleRule('margin-left', '0')

    rerender(
      <FooterContent data-testid="test-id" indent>
        Footer content
      </FooterContent>,
    )

    expect(queryByTestId('test-id')).toHaveStyleRule('margin-left', '28px')
  })
})
