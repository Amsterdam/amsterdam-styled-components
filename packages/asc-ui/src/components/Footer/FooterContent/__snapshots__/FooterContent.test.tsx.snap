// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterContent should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 0;
}

<div
  class="c0"
>
  FooterContent
</div>
`;
